presto-native-execution/presto_cpp/main/types/tests/FunctionMetadataTest.cpp

```diff
@@ -0,0 +1,168 @@
+    EXPECT_EQ(
+        routineCharacteristics.at("determinism"),
+        protocol::Determinism::DETERMINISTIC);
+    EXPECT_EQ(
+        routineCharacteristics.at("nullCallClause"),
+        protocol::NullCallClause::CALLED_ON_NULL_INPUT);
+
+    auto aggregateMetadata = jsonAtIdx.at("aggregateMetadata");
+    EXPECT_EQ(aggregateMetadata.at("intermediateType"), "bigint");
+    EXPECT_EQ(aggregateMetadata.at("isOrderSensitive"), true);
+  }
+}
+
+TEST(FunctionMetadataTest, testSum) {
+  aggregate::prestosql::registerAllAggregateFunctions("");
+  std::string functionName = "sum";
+  json jsonMetadata;
+  getJsonMetadataForFunction(functionName, jsonMetadata);
+  json j = jsonMetadata.at(functionName);
+  const auto numSignatures = 7;
+  EXPECT_EQ(j.size(), numSignatures);
+
+  for (auto i = 0; i < numSignatures; i++) {
+    json jsonAtIdx = j.at(i);
+    EXPECT_EQ(jsonAtIdx.at("functionKind"), protocol::FunctionKind::AGGREGATE);
+    EXPECT_EQ(jsonAtIdx.at("docString"), functionName);
+    EXPECT_EQ(jsonAtIdx.at("schema"), "default");
+    auto outputType = jsonAtIdx.at("outputType");
+    VELOX_CHECK(
+        ((outputType == "real") || (outputType == "double") ||
+         (outputType == "DECIMAL(38,a_scale)") || (outputType == "bigint")));
+    auto paramTypes = jsonAtIdx.at("paramTypes");
+    VELOX_CHECK(
+        ((paramTypes == std::vector<std::string>({"real"})) ||
+         (paramTypes == std::vector<std::string>({"double"})) ||
+         (paramTypes ==
+          std::vector<std::string>({"DECIMAL(a_precision,a_scale)"})) ||
+         (paramTypes == std::vector<std::string>({"tinyint"})) ||
+         (paramTypes == std::vector<std::string>({"smallint"})) ||
+         (paramTypes == std::vector<std::string>({"integer"})) ||
+         (paramTypes == std::vector<std::string>({"bigint"}))));
+
+    auto routineCharacteristics = jsonAtIdx.at("routineCharacteristics");
+    EXPECT_EQ(
+        routineCharacteristics.at("language"), protocol::Language({"CPP"}));
+    EXPECT_EQ(
+        routineCharacteristics.at("determinism"),
+        protocol::Determinism::DETERMINISTIC);
+    EXPECT_EQ(
+        routineCharacteristics.at("nullCallClause"),
+        protocol::NullCallClause::CALLED_ON_NULL_INPUT);
+
+    auto aggregateMetadata = jsonAtIdx.at("aggregateMetadata");
+    auto intermediateType = aggregateMetadata.at("intermediateType");
+    VELOX_CHECK(
+        (intermediateType == "double") || (intermediateType == "VARBINARY") ||
+        (intermediateType == "bigint"));
+    EXPECT_EQ(aggregateMetadata.at("isOrderSensitive"), true);
+  }
+}
+
+TEST(FunctionMetadataTest, testRank) {
+  window::prestosql::registerAllWindowFunctions("");
+  std::string functionName = "rank";
+  json jsonMetadata;
+  getJsonMetadataForFunction(functionName, jsonMetadata);
+  json j = jsonMetadata.at(functionName);
+  const auto numSignatures = 1;
+  EXPECT_EQ(j.size(), numSignatures);
+
+  for (auto i = 0; i < numSignatures; i++) {
```

Comment:
Why is there a loop here when there is only one signature.? It seems odd.

Reply:
It was just used to be consistent with the other tests and since the `getFunctionMetadata` returns an array but removed it now, thanks for pointing it out.